sktime/datasets/generators.py

```diff
@@ -12,21 +12,37 @@
 
 from numpy import ndarray
 from pandas import Series, DataFrame
+from sklearn.base import BaseEstimator
+
 
 __all__ = ["ArmaGenerator", "LinearGenerator", "NoiseGenerator"]
 
 
-class Generator(ABC):
+class Generator(BaseEstimator):
     """
-    Abstrct class for generators.
+    A base class for generators.
+
+    Notes
+    -----
+    All generator parameters should be specified as keyword arguments in
+    their `__init__`.
+    All generators should implement the unified interface given below.
+
     """
 
     @abstractmethod
-    def sample(self, n_sample):
+    def sample(self, n_sample, n_instance):
```

Comment:
```suggestion
    def sample(self, n_instances=None, n_variables=None, n_timepoints=None):
```
with some useful default values

Alternatively we could distinguish between generators for different kinds of data (time series and panel data)?